functions/pipeline.py

```diff
@@ -0,0 +1,467 @@
+
+    # Create VAE directories
+    output_dirs = [os.path.join(base_dir, dataset_name, "models"),
+                   os.path.join(base_dir, dataset_name, "logs")]
+
+    # Check if analysis output directory exist otherwise create
+    for each_dir in output_dirs:
+        if os.path.exists(each_dir) == False:
+            print('creating new directory: {}'.format(each_dir))
+            os.makedirs(each_dir, exist_ok=True)
+
+        # Check if NN architecture directory exist otherwise create
+    for each_dir in output_dirs:
+        new_dir = os.path.join(each_dir, train_architecture)
+        if os.path.exists(new_dir) == False:
+            print('creating new directory: {}'.format(new_dir))
+            os.makedirs(new_dir, exist_ok=True)
+
+    # Create results directories
+    output_dirs = [os.path.join(base_dir, dataset_name, "results")]
+
+    # Check if analysis output directory exist otherwise create
+    for each_dir in output_dirs:
+        if os.path.exists(each_dir) == False:
+            print('creating new directory: {}'.format(each_dir))
+            os.makedirs(each_dir, exist_ok=True)
+
+    # Check if 'saved_variables' directory exist otherwise create
+    for each_dir in output_dirs:
+        new_dir = os.path.join(each_dir, 'saved_variables')
+
+        if os.path.exists(new_dir) == False:
+            print('creating new directory: {}'.format(new_dir))
+            os.makedirs(new_dir, exist_ok=True)
+
+    # Create local directories to store intermediate files
+    output_dirs = [os.path.join(local_dir, "experiment_simulated"),
+                   os.path.join(local_dir, "partition_simulated")]
+
+    # Check if analysis output directory exist otherwise create
+    for each_dir in output_dirs:
+        if os.path.exists(each_dir) == False:
+            print('creating new directory: {}'.format(each_dir))
+            os.makedirs(each_dir, exist_ok=True)
+
+
+def normalize_expression_data(base_dir,
+                              config_file,
+                              raw_input_data_file):
+    '''
+    Normalize the expression data
+
+    Arguments
+    ----------
+    base_dir: str
+        Root directory containing analysis subdirectories
+
+    config_file: str
+        File containing user defined parameters
+
+    raw_input_data_file: str
+        File containing raw expression data
+    '''
+
+    # Read in config variables
+    params = utils.read_config(config_file)
+
+    # Load parameters
+    dataset_name = params['dataset_name']
+
+    # Read data
+    rpkm_data = pd.read_table(
+        raw_input_data_file,
+        header=0,
+        sep='\t',
+        index_col=0)
+
+    print(rpkm_data.shape)
+
+    # 0-1 normalize per gene
+    rnaseq_scaled_df = preprocessing.MinMaxScaler().fit_transform(rpkm_data)
+    rnaseq_scaled_df = pd.DataFrame(rnaseq_scaled_df,
+                                    columns=rpkm_data.columns,
+                                    index=rpkm_data.index).T
```

Comment:
Just making sure, you have genes as column names and samples as rows, right? Otherwise the normalization will be per-sample instead of per-gene

Reply:
Thank you for checking this.  That is correct, the normalization is performed per gene.  The reason for the transpose is because the downstream functions are expected to get a dataset that is gene x sample.  I'm not sure of a good way to check the dimension without consistent sample and gene naming.  There were earlier bugs when the dimensions were switched accidentally